crates/orchestrator/src/jobs/da_job/mod.rs

```diff
@@ -180,36 +348,79 @@ mod tests {
         assert_eq!(da_word, expected);
     }
 
-    mod test_state_update_to_blob_data {
-        use super::*;
-        use serde_json::Error;
-        use std::fs::{read_to_string, File};
-        use std::io::{self, BufRead};
+    fn write_biguint_to_file(numbers: &Vec<BigUint>, file_path: &str) {
+        let mut file = File::create(file_path).expect("not able to create the files");
+        for number in numbers {
+            writeln!(file, "{}", number).expect("unable to write to the file");
+        }
+    }
+
+    #[rstest]
+    // #[case(631861, "src/jobs/test_utils/test_blob_631861.txt")]
+    // #[case(638353, "src/jobs/test_utils/test_blob_638353.txt")]
+    // #[case(639404, "src/jobs/test_utils/test_blob_639404.txt")]
+    #[case(640641, "src/jobs/test_utils/test_blob_640641.txt")]
+    #[tokio::test]
+    async fn test_state_update_to_blob_data(#[case] block_no: u64, #[case] file_path: &str) {
+        // @note: not yet done
+        let config = init_config(
+            Some(format!("https://starknet-mainnet.infura.io/v3/3753abe17c124d088f4e68d58f257452")),
```

Comment:
let's store the state update json locally. not a good practice to rely on external API calls for unit tests

Reply:
how will fetch the nonce for the addresses where state_diff is non-zero? using mocks?

Reply:
ya, we would need to mock the rpc endpoint like we do in some test currently using httpmock

Reply:
should I store the nonce values for the test case?